[helm] found character '\t' that cannot start any token

<!-- Please keep the structure below, or your issue may be closed. -->

**Describe the bug**

The Helm template contains a trailing tab character which is not compliant with the [YAML spec](http://yaml.org/spec/1.2/spec.html#id2777534).

https://github.com/grafana/loki/blob/main/production/helm/loki/templates/single-binary/statefulset.yaml#L179C21-L179C22

**To Reproduce**
Steps to reproduce the behavior:

```
❯ helm template loki production/helm/loki/ --set 'singleBinary.replicas=1' | yamllint - | grep 'syntax error'
  954:21    error    syntax error: found character '\t' that cannot start any token (syntax)
```

**Expected behavior**

No trailing tab.

**Environment:**

```
❯ helm version                                                                                               
version.BuildInfo{Version:"v3.13.2", GitCommit:"v3.13.2", GitTreeState:"", GoVersion:"go1.21.4"}
```

```
❯ yamllint -v     
yamllint 1.32.0
```

**Screenshots, Promtail config, or terminal output**

Screenshot from `helm template` output.

![Screenshot 2023-12-05 at 09 52 00](https://github.com/grafana/loki/assets/4059732/e9ec1a13-1314-4106-95fb-377ae6e6d38b)

--- a/production/helm/loki/CHANGELOG.md
+++ b/production/helm/loki/CHANGELOG.md
@@ -13,6 +13,10 @@ Entries should include a reference to the pull request that introduced the chang
 
 [//]: # (<AUTOMATED_UPDATES_LOCATOR> : do not remove this line. This locator is used by the CI pipeline to automatically create a changelog entry for each new Loki release. Add other chart versions and respective changelog entries bellow this line.)
 
+## 5.42.2
+
+- [BUGFIX] Remove trailing tab character in statefulset templates
+
 ## 5.42.1
 
 - [BUGFIX] Added missing annotations to loki-read StatefulSet.

--- a/production/helm/loki/Chart.yaml
+++ b/production/helm/loki/Chart.yaml
@@ -3,7 +3,7 @@ name: loki
 description: Helm chart for Grafana Loki in simple, scalable mode
 type: application
 appVersion: 2.9.4
-version: 5.42.1
+version: 5.42.2
 home: https://grafana.github.io/helm-charts
 sources:
   - https://github.com/grafana/loki

--- a/production/helm/loki/README.md
+++ b/production/helm/loki/README.md
@@ -1,6 +1,6 @@
 # loki
 
-![Version: 5.42.1](https://img.shields.io/badge/Version-5.42.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 2.9.4](https://img.shields.io/badge/AppVersion-2.9.4-informational?style=flat-square)
+![Version: 5.42.2](https://img.shields.io/badge/Version-5.42.2-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 2.9.4](https://img.shields.io/badge/AppVersion-2.9.4-informational?style=flat-square)
 
 Helm chart for Grafana Loki in simple, scalable mode
 

--- a/production/helm/loki/templates/backend/statefulset-backend.yaml
+++ b/production/helm/loki/templates/backend/statefulset-backend.yaml
@@ -263,7 +263,7 @@ spec:
         {{- end }}
   {{- if .Values.backend.persistence.volumeClaimsEnabled }}
   volumeClaimTemplates:
-    - apiVersion: v1	
+    - apiVersion: v1
       kind: PersistentVolumeClaim
       metadata:
         name: data

--- a/production/helm/loki/templates/read/statefulset-read.yaml
+++ b/production/helm/loki/templates/read/statefulset-read.yaml
@@ -175,7 +175,7 @@ spec:
         {{- toYaml . | nindent 8 }}
         {{- end }}
   volumeClaimTemplates:
-    - apiVersion: v1	
+    - apiVersion: v1
       kind: PersistentVolumeClaim
       metadata:
         name: data

--- a/production/helm/loki/templates/single-binary/statefulset.yaml
+++ b/production/helm/loki/templates/single-binary/statefulset.yaml
@@ -176,7 +176,7 @@ spec:
         {{- end }}
   {{- if .Values.singleBinary.persistence.enabled }}
   volumeClaimTemplates:
-    - apiVersion: v1	
+    - apiVersion: v1
       kind: PersistentVolumeClaim
       metadata:
         name: storage

